### PR TITLE
fix: fix duplicate message issue for login

### PR DESCRIPTION
**Pull request by SLdragon**
Refer this: https://github.com/OfficeDev/Microsoft-Teams-Samples/blob/main/samples/bot-conversation-sso-quickstart/csharp_dotnetcore/TokenExchangeHelper.cs

### CODE REVIEW

**Review by YitongFeng-git — APPROVED** [review]

File: bot/js/bots/teamsBot.js [hunk 1]
```diff
@@ -10,9 +12,9 @@ class TeamsBot extends DialogBot {
      * @param {UserState} userState
      * @param {Dialog} dialog
      */
-    constructor(conversationState, userState, dialog) {
+    constructor(conversationState, userState, dialog, storage) {
```
Comment (adashen) [inline bot/js/bots/teamsBot.js:15]: shall we directly use storage or we should leverage conversation state?

Comment (SLdragon) [reply bot/js/bots/teamsBot.js:15]: Sure, will investigate the conversationState

**Review by dilin-MS — APPROVED** [review]

File: bot/js/bots/teamsBot.js (hunk 1 above)
Comment (blackchoey) [reply bot/js/bots/teamsBot.js:15]: We also need to consider deleting the state after the SSO finished. Not sure whether conversation state already has such features by default.

File: bot/js/index.js [hunk 2]
```diff
@@ -55,7 +55,7 @@ const userState = new UserState(memoryStorage);
 // Create the main dialog.
 const dialog = new MainDialog();
 // Create the bot that will handle incoming messages.
-const bot = new TeamsBot(conversationState, userState, dialog);
+const bot = new TeamsBot(conversationState, userState, dialog, memoryStorage);
```
Comment (blackchoey) [inline bot/js/index.js:58]: Also need to tell developers that they need a distributed storage if the application runs on multiple servers. Can you help confirm whether we want to add this information as code comment, or in our document?

Comment (SLdragon) [reply bot/js/index.js:58]: Sure, this message has been added before should dedup function

File: bot/js/bots/teamsBot.js (hunk 1 above)
Comment (SLdragon) [reply bot/js/bots/teamsBot.js:15]: OK

**Pull request merged**